src/resolver/didValidator.ts

```diff
@@ -143,25 +147,67 @@ async function resolvePermissionFromService(service: Service, did: string): Prom
 }
 
 /**
- * Extracts the Trust Registry base URL and the schema ID from a given schema `refUrl`.
+ * Resolves and validates a W3C Verifiable Credential by extracting and verifying
+ * the issuer's DID and evaluating the credential against the configured trust registries.
+ *
+ * @param cred   The W3C Verifiable Credential to be resolved and assessed.
+ * @param options Configuration object containing the DID resolver and the set
+ *                of verifiable public registries used during trust evaluation.
  *
- * Example:
- * Input:  "https://registry.example.com/schemas/v1/1"
- * Output: {
- *   trustRegistry: "https://registry.example.com/schemas",
- *   schemaId: "1"
- * }
+ * @returns A TrustResolution object containing the issuer's DID Document,
+ *          the verification outcome, and any associated service information.
+ */
+export async function resolveCredential(
+  cred: W3cVerifiableCredential,
+  options: ResolverConfig,
+): Promise<TrustResolution> {
```

Comment:
I still don't like this "TrustResolution" as return type. We need to use a different return type for this method to avoid confusions: we are not sending any DID Document and also we don't have a "service" and "serviceProvider"; maybe we can have a "issuer" if we want, but it is not neccesarily a Service (e.g. an Org can be the issuer of a credential) so it is weird to call it (and type it as a) service.